test/integration/server-selection/readpreference.test.ts

```diff
@@ -463,4 +457,214 @@ describe('ReadPreference', function () {
+              expect(events[0]).to.containSubset({
+                commandName: 'find',
+                command: {
+                  $readPreference: { mode: 'primaryPreferred' }
+                }
+              });
+
+              checkedPrimary = true;
+            }
+
+            await client.close();
+          }
+
+          expect(checkedPrimary).to.be.equal(true);
+        }
+      });
+
+      it('should not attach a read preference to the read command for standalone', {
+        metadata: { requires: { topology: 'single', mongodb: '>=3.6' } },
+        test: async function () {
+          const client = this.configuration.newClient(
+            {
+              readPreference: 'primary',
+              directConnection: true
+            },
+            {
+              monitorCommands: true
+            }
+          );
+          events = [];
+          client.on('commandStarted', event => {
+            if (event.commandName === 'find') {
+              events.push(event);
+            }
+          });
+          let serverError;
+          try {
+            await client.db('test').collection('test').findOne({ a: 1 });
+          } catch (error) {
+            serverError = error;
+          }
+
+          expect(serverError).to.not.exist;
+          expect(events[0]).to.not.containSubset({
+            commandName: 'find',
+            command: {
+              $readPreference: { mode: 'primaryPreferred' }
+            }
+          });
+          expect(events[0]).to.not.containSubset({
+            commandName: 'find',
+            command: {
+              $readPreference: { mode: 'primary' }
+            }
+          });
+
+          await client.close();
+        }
+      });
+    });
+
+    context('when readPreference is secondary', () => {
+      it('should attach a read preference of secondary to the read command for replicaset', {
+        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
+        test: async function () {
+          let checkedSecondary = false;
+
+          for (const server of this.configuration.options.hostAddresses) {
+            const { host, port } = server.toHostPort();
+            const client = this.configuration.newClient(
+              {
+                readPreference: 'secondary',
+                directConnection: true,
+                host,
+                port
+              },
+              {
+                monitorCommands: true
+              }
+            );
+            events = [];
+            client.on('commandStarted', event => {
+              if (event.commandName === 'find') {
+                events.push(event);
+              }
+            });
+            let serverStatus;
+
+            try {
+              const admin = client.db().admin();
+              serverStatus = await admin.serverStatus();
+            } catch (serverStatusError) {
+              expect(serverStatusError).to.not.exist;
+            }
```

Comment:
We could just let the serverStatus function throw to fail the test instead of this assertion right?